server/controllers/common.js

```diff
@@ -4,11 +4,34 @@ const pageHelper = require('../helpers/page')
 const _ = require('lodash')
 const CleanCSS = require('clean-css')
 const moment = require('moment')
+const acceptLanguageParser = require('accept-language-parser')
 
 /* global WIKI */
 
 const tmplCreateRegex = /^[0-9]+(,[0-9]+)?$/
 
+/**
+ * Calculates the best language by `Accept-Language` header of request and
+ * available locales of pages with specified path.
+ * @param {express.Request} req
+ * @param {*} pageArgs
+ * @returns
+ */
+async function findBestLocaleFor(req, pageArgs) {
```

Comment:
I don't know where is the best place for such a function so I write here. If there's better place, Could you let me know?